File: mysql-explain-bargraph.pl
#!/usr/bin/perl -w
#
# mysql-explain-bargraph.pl - Generate horizontal bar chart SVG from MySQL EXPLAIN ANALYZE JSON
#
# Thin wrapper: uses unified parser in mysql-explain.pl with --type bargraph.
# Usage: ./mysql-explain-bargraph.pl [options] explain.json > output.svg
#
use strict;
use File::Basename;
use File::Spec;

my $script_dir = dirname(File::Spec->rel2abs($0));
my $unified = File::Spec->catfile($script_dir, "mysql-explain.pl");
die "Cannot find mysql-explain.pl in $script_dir\n" unless -x $unified;

exec($unified, "--type", "bargraph", @ARGV) or die "Cannot exec $unified: $!\n";

File: mysql_explain_bargraph.py
#!/usr/bin/env python3
"""Thin wrapper: bar chart from MySQL EXPLAIN JSON. Runs unified CLI with --type bargraph."""
import sys
sys.argv = [sys.argv[0], "--type", "bargraph"] + sys.argv[1:]
if __name__ == "__main__":
    from myflames.cli import main
    main()
